barretenberg/cpp/src/barretenberg/eccvm_recursion/eccvm_recursive_verifier.cpp

```diff
@@ -31,7 +31,11 @@ template <typename Flavor> void ECCVMRecursiveVerifier_<Flavor>::verify_proof(co
     VerifierCommitments commitments{ key };
     CommitmentLabels commitment_labels;
 
-    const auto circuit_size = transcript->template receive_from_prover<BF>("circuit_size");
+    // TODO(https://github.com/AztecProtocol/barretenberg/issues/1040): Extract circuit size as BF (field_t) then
+    // convert to FF (bigfield fq) since this is what's expected by ZM. See issue for more details.
```

Comment:
I think what needs to change is ZM. ZM shouldn't take in an FF, but rather a Builder::FF in the stdlib case. In the native case, it should be uint32_t.

Reply:
Updated the issue to reflect this possibility